compiler/modules/types/listProj.bal

```diff
@@ -67,39 +68,44 @@ function listProjPath(Context cx, int k, Conjunction? pos, Conjunction? neg) ret
             rest = NEVER;
         }
     }
-    return listProjExclude(cx, k, members, rest, neg);
+    return listProjExclude(cx, k, members, rest, listConjunction(cx, neg));
 }
 
 // Precondition k >= 0 and members[i] not empty for all i
 // This finds the projection of e[k], excluding the list of atoms in neg
 // when the type of e is given by members and rest.
 // Based on listInhabited
 // Corresponds to phi^x in AMK tutorial generalized for list types.
-function listProjExclude(Context cx, int k, FixedLengthArray members, SemType rest, Conjunction? neg) returns SemType {
+function listProjExclude(Context cx, IntSubtype? k, FixedLengthArray members, SemType rest, ListConjunction? neg) returns SemType {
     if neg == () {
-        return listMemberAt(members, rest, k);
+        return listAtomicMemberTypeAt(members, rest, k);
     }
     else {
         int len = members.fixedLength;
-        ListAtomicType nt = cx.listAtomType(neg.atom);
+        ListAtomicType nt = neg.listType;
         int negLen = nt.members.fixedLength;
+        SemType p = NEVER;
         if len < negLen {
             if isNever(rest) {
                 return listProjExclude(cx, k, members, rest, neg.next);
             }
+            foreach int i in members.initial.length() + 1 ..< nt.members.initial.length() {
```

Comment:
There seems to be a chunk of very similar code in listInhabited, which I don't understand either.